ReverseWordsInAString.java

```diff
@@ -0,0 +1,24 @@
+class ReverseWordsInAString {
+    public String reverseWords(String s) {
+        // remove leading spaces
+        s = s.trim();
+        // split by multiple spaces
+        List<String> wordList = Arrays.asList(s.split("\\s+"));
+        // reverse the list and join
+        Collections.reverse(wordList);
+        return String.join(" ", wordList);
+    }
+
+    public static void main(final String[] args) {
+        // here the time complexity would be O(n) as we are traversing through the string only once

```

Comment:
Aren't you traversing 3 times?  Once to split, once to re-join, and possibly once to reverse?  The last one is done at the word level so it's a bit weird but should be O(n), since reversing something can be done in O(n) and each word is composed of at least one character.